client/app/api.js

```diff
@@ -0,0 +1,4 @@
+export function getTree(path) {
+  let url = 'http://localhost:8080/api/tree/'; 
+  return fetch(url += path).then(res => res.json());
```

Comment:
Every call to `getTree` modifies url via +=. Use either + or just template literal w/ interpolation